### PR TITLE
Add configs to enable/disable different anomaly detectors individually.

### PR DESCRIPTION
**Pull request by efeg**

### CODE REVIEW

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SelfHealingNotifierTest.java [hunk 1]
```diff
@@ -43,83 +45,111 @@ public void testOnBrokerFailure() {
...
-    final long failureTime1 = 200L;
-    final long failureTime2 = 400L;
     final long startTime = 500L;
     Time mockTime = new MockTime(startTime);
     KafkaCruiseControl mockKafkaCruiseControl = EasyMock.mock(KafkaCruiseControl.class);
     TestingBrokerFailureAutoFixNotifier anomalyNotifier = new TestingBrokerFailureAutoFixNotifier(mockTime);
-    anomalyNotifier.configure(Collections.singletonMap(SelfHealingNotifier.SELF_HEALING_ENABLED_CONFIG, "false"));
 
+    Map<String, String> selfHealingExplicitlyDisabled = new HashMap<>(3);
+    selfHealingExplicitlyDisabled.put(SelfHealingNotifier.SELF_HEALING_BROKER_FAILURE_ENABLED_CONFIG, "false");
```
Comment (kidkun) [inline cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SelfHealingNotifierTest.java]: I think we can set `SelfHealingNotifierSELF_HEALING_ENABLED_CONFIG` to `True` here to verify the overriding of  specific config over general config.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SelfHealingNotifier.java [hunk 2]
```diff
@@ -79,30 +91,32 @@ public AnomalyNotificationResult onBrokerFailure(BrokerFailures brokerFailures)
...
    */
-  public void alert(Object anomaly, boolean autoFixTriggered, long selfHealingStartTime) {
-    if (_selfHealingEnabled) {
-      LOG.warn("Anomaly detected {}, plan self-healing start time {}.", anomaly, toDateString(selfHealingStartTime));
-    } else {
-      LOG.warn("Anomaly detected {}. Self healing is disabled.", anomaly);
-    }
+  public void alert(Object anomaly, boolean autoFixTriggered, long selfHealingStartTime, AnomalyType anomalyType) {
+    LOG.warn("{} detected {}. Self healing {}.", anomalyType, anomaly,
+             _selfHealingMetricAnomalyEnabled ? String.format("start time %s", toDateString(selfHealingStartTime))
```
[the remark this answers is not in the document]
Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SelfHealingNotifier.java]: We should indeed check for whether the corresponding anomaly type is enabled -- note that this is different from `autoFixTriggered `.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SelfHealingNotifier.java [hunk 3]
```diff
@@ -79,30 +91,32 @@ public AnomalyNotificationResult onBrokerFailure(BrokerFailures brokerFailures)
...
+                                              : "is disabled");
+
     if (autoFixTriggered) {
       LOG.warn("Self-healing has been triggered.");
```
[the remark this answers is not in the document]
Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SelfHealingNotifier.java]: Please see my response to the previous feedback above.

[the remark this answers is not in the document]
Comment (kidkun) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SelfHealingNotifier.java]: I see, sorry I mess the `enable` and `triggered` concept.

**Comment by kidkun** [comment]
LGTM.

**Pull request merged**